aws_lambda_builders/workflows/rust_cargo/actions.py

```diff
@@ -0,0 +1,148 @@
+"""
+Rust Cargo build actions
+"""
+
+import logging
+import os
+
+from aws_lambda_builders.workflow import BuildMode
+from aws_lambda_builders.actions import ActionFailedError, BaseAction, Purpose
+from aws_lambda_builders.architecture import X86_64, ARM64
+from .exceptions import CargoLambdaExecutionException
+from .utils import OSUtils
+
+
+LOG = logging.getLogger(__name__)
+
+
+class RustCargoLambdaBuildAction(BaseAction):
+    NAME = "CargoLambdaBuild"
+    DESCRIPTION = "Building the project using Cargo Lambda"
+    PURPOSE = Purpose.COMPILE_SOURCE
+
+    def __init__(
+        self,
+        source_dir,
+        binaries,
+        mode,
+        subprocess_cargo_lambda,
+        architecture=X86_64,
+        handler=None,
+        flags=None,
+    ):
+        """
+        Build the a Rust executable
+
+        :type source_dir: str
+        :param source_dir:
+            Path to a folder containing the source code
+
+        :type binaries: dict
+        :param binaries:
+            Resolved path dependencies
+
+        :type mode: str
+        :param mode:
+            Mode the build should produce
+
+        :type architecture: str, optional
+        :param architecture:
+            Target architecture to build the binary, either arm64 or x86_64
+
+        :type handler: str, optional
+        :param handler:
+            Handler name in `bin_name` format
+
+        :type flags: list, optional
+        :param flags:
+            Extra list of flags to pass to `cargo lambda build`
+
+        :type subprocess_cargo_lambda: aws_lambda_builders.workflows.rust_cargo.cargo_lambda.SubprocessCargoLambda
+        :param subprocess_cargo_lambda: An instance of the Cargo Lambda process wrapper
+        """
+
+        self._source_dir = source_dir
+        self._mode = mode
+        self._binaries = binaries
+        self._handler = handler
+        self._flags = flags
+        self._architecture = architecture
+        self._subprocess_cargo_lambda = subprocess_cargo_lambda
+
+    def build_command(self):
+        cmd = [self._binaries["cargo"].binary_path, "lambda", "build"]
+        if self._mode == BuildMode.RELEASE:
+            cmd.append("--release")
```

Comment:
Can we make this the default behavior and only remove this flag is `self._mode` is `BuildMode.DEBUG`? 
Reason is building without `--release` seems to produce a much larger artifact that exceeds Lambda's size limit. Defaulting it to build with `--release` seems to cause less deployment issue

Reply:
FYI, Go and Dotnet workflows also default to `BuildMode.RELEASE`:
- https://github.com/aws/aws-lambda-builders/blob/2ad17ffb6f4a81f82b458614f2cf7c992610f2e1/aws_lambda_builders/workflows/go_modules/builder.py#L50-L52
- https://github.com/aws/aws-lambda-builders/blob/2ad17ffb6f4a81f82b458614f2cf7c992610f2e1/aws_lambda_builders/workflows/dotnet_clipackage/actions.py#L87-L89